### DISCUSSION TITLE
Question about displaying Envelope-from

### DISCUSSION THREAD

**Discussion by Toni202012569** [postalserver/postal#2686] [Help with using Postal]
Good afternoon.

If I understand correctly, the Envelope-from field should contain the sender's address. However, when sending via Postal, I see that this line indicates a record generated based on the CNAME, which is probably not entirely correct envelope-from=wwusqx@psrp.domen.com;. I tested sending through servers from other providers and found that in all letters, in this line the sender's address is indicated correctly envelope-from=info@domen.com; Has anyone encountered this before and how to bring the envelope-from entry into compliance with the sender's email.

**Comment by willpower232** [comment]
It is definitely a complicated subject, the psrp is mostly used to handle bounces (and in some cases verify the legitimacy of the email) so I think it is "fine" as it is but definitely confusing as you say. I believe some email providers (gmail) show this on screen which doesn't make a lot of sense to the end user.

I don't know if applying the sender header when creating the message is enough to make it look better to the receipient?
